src/github.ts

```diff
@@ -0,0 +1,140 @@
+  core.info(`Requesting Reviewers for PR #${pr} to: ${reviewers.join(',')}`);
+
+  const response = await client.rest.pulls.requestReviewers({
+    owner: github.context.repo.owner,
+    repo: github.context.repo.repo,
+    pull_number: pr,
+    reviewers: reviewers,
+  });
+  coreDebugJson(response, `requestReviewers(${pr}, [${reviewers.join(',')}]) > response`);
+}
+
+/**
+ *
+ * Assign the current context's PR to the specified assignees.
+ * @param assignees The list of assignees to assign the PR.
+ */
+export async function setAssignees(assignees: string[]): Promise<void> {
+  const pr = getPr().number;
+  core.info(`Setting Assignees for PR #${pr} to: ${assignees.join(',')}`);
+
+  const response = await client.rest.issues.addAssignees({
+    owner: github.context.repo.owner,
+    repo: github.context.repo.repo,
+    issue_number: pr,
+    assignees: assignees,
+  });
+  coreDebugJson(response, `setAssignees(${pr}, [${assignees.join(',')}]) > response`);
+}
+
+export type CheckRun = components['schemas']['check-run'];
+
+/**
+ * Get the checkruns for the current context's workflow_run for the head SHA.
+ */
+export async function getCheckRuns(): Promise<CheckRun[]> {
+  const ref = github.context.payload.workflow_run.head_sha;
+  const { owner, repo } = github.context.repo;
+
+  core.info(`Retrieving check runs for ${owner}/${repo}@${ref}...`);
+  const checkRuns: CheckRun[] = (
+    await client.rest.checks.listForRef({
+      owner,
+      repo,
+      ref,
+    })
+  ).data.check_runs;
+
+  core.info(`Retrieved ${checkRuns.length} check runs.`);
+  coreDebugJson(checkRuns, 'getCheckRuns() > checkRuns');
+
+  return checkRuns;
+}
+
+/**
+ * Get the required checks for the base ref.
+ */
+export async function getRequiredCheckNames(): Promise<string[] | undefined> {
+  const baseRef = getPr()?.base.ref;
+  core.info(`Base Ref: ${baseRef}`);
+  if (baseRef === undefined) {
+    core.error(`Error getting base ref for PR #${getPr()?.number}.`);
+    return undefined;
+  }
+  const { owner, repo } = github.context.repo;
+
+  core.info(`Retrieving branch protection information for ${owner}/${repo}@${baseRef}...`);
+  const result: components['schemas']['status-check-policy'] = (
+    await client.rest.repos.getStatusChecksProtection({
+      owner,
+      repo,
+      branch: baseRef,
+    })
+  ).data;
+
+  coreDebugJson(result, 'getRequiredChecks() > result');
+
+  return result?.contexts;
+}
+
+/**
+ * Get user inputs as an array (expects the user input to be CSV)
+ * @param name The name of the user input
+ * @param defaultVal The default value
+ */
+export function getInputArray(name: string, defaultVal: string[]): string[] {
+  return (
+    core
+      .getInput(name)
+      .split(',')
+      .map((i) => i.trim()) || defaultVal
+  );
+}
+
+/**
+ * Get user inputs
+ * @param name The name of the user input
+ * @param defaultVal The default value
+ */
+export function getInput(name: string, defaultVal: string | null): string | null {
```

Comment:
```suggestion
export function getInputWithDefault(name: string, defaultValue?: string | null): string | null {
```

Reply:
I'm not sure why you'd allow null to be passed as a default value. Can you elaborate?

Reply:
The value should never be null - if the user configures this in their workflow, `core.getInput()` will pull the value the user specified; if the user does **not** configure this in their workflow, the `action.yaml` file specifies the default to be `600` so that will be used.

Now that I'm looking at this more closely, I actually realize this is a lingering bug 🐛 - my intent is for a user to be able to turn 'off' this functionality, but that's not currently possible (due to how I was handling this as a string until it was needed as a number; `'0'` is not falsely, thus a bug existed). I've just updated this to fix.